libs/ui/lib/modal/Modal.tsx

```diff
@@ -126,6 +127,7 @@ Modal.Footer = ({
   onAction: () => void
   actionType?: 'primary' | 'danger'
   actionText: React.ReactNode
+  actionLoading?: boolean
   cancelText?: string
   disabled?: boolean
```

Comment:
Adding even more props here to pass through to the children is a clear sign this should be refactored into a compositional API.